documentation/wiki/Logging-Internals.md

```diff
@@ -0,0 +1,141 @@
+Example of forwarders:
+  * `CentralForwardingLogger`
+  * `ConfigurableForwardingLogger`
+* Each forwarder process the trace stream and forwards to an `EventRedirectorToSink` (1 redirector per 1 forwarder)
+* `EventRedirectorToSink` forwards to `EventSourceSink` (the instance of `EventSourceSink` was passed to `EventRedirectorToSink` during its construction). There is 1 `EventSourceSink` per each `EventRedirectorToSink`
+* `EventSourceSink` distributes events to registered Loggers. There can be multiple registered Loggers to the single `EventSourceSink` (this is the case when Logger was registered as 'Central Logger' via [`RegisterLogger`](#RegisterLogger)).
+
+
+![In node logging](InNodeLoggingFlow.png)
+
+The above diagram is simplified (it doesn't capture the calls within a single component), the current state of calls (that is subject to changes) is illustrated by the following stack trace capture:
+
+![In node logging - stack](InProcLoggingStack.png)
+
+ ### Delivery of events in a distributed build
+
+ In case of distributed execution of build, NodeManager is requesting execution of additional Out Of Process (OOP) Nodes. Logging events from those nodes are transferred to the Entrypoint node, where they are eventually consumed by the Central Loggers. In addition to the OOP Nodes, build can have a single 'In Proc Node' - an execution node sharing the same process with the Entrypoint Node. For this reason the described process and diagram for the [Delivery of events in a single node build](#delivery-of-events-in-a-single-node-build) can apply fully for the distributed build as well (in addition to the following).
+
+ #### OOP Node 
+
+The initial part of the workflow of logging is identical to the in-proc delivery:
+
+* There is 1 instance of LoggingService per build.
+* It holds 1 instance of EventSourceSink pumping events to forwarders.
+* There can be multiple forwarders (one shared for all central loggers registered via [`RegisterLogger`](#RegisterLogger) and one per each distributed logger registered via [`RegisterDistributedLogger`](#RegisterDistributedLogger)).
+Example of forwarders:
+  * `CentralForwardingLogger`
+  * `ConfigurableForwardingLogger`
+* Each forwarder processes the trace stream and forwards to an `EventRedirectorToSink` (1 redirector per 1 forwarder)
+
+Processing from `EventRedirectorToSink` differs. This is given by injection of a single sink `BuildEventArgTransportSink` (injected via [`InitializeNodeLoggers`](#InitializeNodeLoggers)) as a consumer of `EventRedirectorToSink` data:
+
+* There is single `BuildEventArgTransportSink` per build, that consumes data from all `EventRedirectorToSink`. The data are enriched with the SinkId (AKA LoggerId) - the Id was created during the Logger registration and remoted to the OOP node together with the [`LoggerDescription`](#LoggerDescription).
```

Comment:
Is there a difference betwwe SinkId and LoggerID? Or is it just different names for the same thing?

Reply:
It is the same thing. But different components in code refer to it via those 2 different terms - so I wanted to make it explicit here that it's the same thing